pkg/applicationserver/io/packages/lora-cloud-device-management-v1/api/tokens.go

```diff
@@ -0,0 +1,127 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"go.thethings.network/lorawan-stack/pkg/applicationserver/io/packages/lora-cloud-device-management-v1/api/objects"
+)
+
+// Tokens is an API client for the Token Management API.
+type Tokens struct {
+	cl *Client
+}
+
+const (
+	tokenEntity = "token"
+	nameParam   = "name"
+	renewParam  = "renew"
+)
+
+// List returns the tokens.
+func (t *Tokens) List() ([]objects.TokenInfo, error) {
+	req, err := t.cl.newRequest(http.MethodGet, tokenEntity, "", listOperation, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := t.cl.cl.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var response tokensListResponse
+	err = parse(&response, resp.Body)
```

Comment:
Does this guarantee that it reads all contents from `resp.Body`?

Reply:
Added automatic discard to `parse` in order to deal with this.